framework/decode/dx12_resource_value_mapper.cpp

```diff
@@ -1699,10 +1731,89 @@ void Dx12ResourceValueMapper::GetStateObjectLrsAssociationInfo(
             auto num_exports           = dxil_lib_desc_decoder->GetPointer()->NumExports;
             if (num_exports == 0)
             {
-                // TODO: Parse the names of all shaders exported from the DXIL library.
+#if defined(GFXRECON_DXC_SUPPORT)
+                // If D3D12_DXIL_LIBRARY_DESC::NumExports == 0, everything in the DXIL library is exported. Use
+                // reflection to get the list of exported shader names.
+                HRESULT                         hr;
```

Comment:
There's a lot of machinery here - do we know how it might affect CPU-side perf to do this work?

Wondering if any of these things can be a 1-time init and reusable thing? For instance all `DxcCreateInstance` calls

Reply:
I did not do any specific performance measuring. This code should only run during unoptimized DXR or EI replay where performance is already heavily affected.

The `dxc_utils` object could be reused but I don't know whether that is a heavy-weight operation. The other classes are based on the incoming DXIL library so the code would need to compare the DXIL library's bytes to detect if previous objects or results could be reused.